// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.langchain.ask";
option java_outer_classname = "AskProto";
option objc_class_prefix = "HLW";

package ask;

// The greeting service definition.
service LangchainAsk {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SayHelloStreamReply (HelloRequest) returns (stream HelloReply) {}

  rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}

  rpc LoadGithubRepository(LoadRepoRequest) returns (LoadRepoReply) {}

  rpc EstablishSession(EstablishSessionRequest) returns (EstablishSessionReply) {}

  rpc processUserQuestion(EstablishSessionRequest) returns (EstablishSessionReply) {}
}

message LoadRepoRequest{
  string name = 1;
}

message LoadRepoReply{
  bool success = 1;
  string message = 2;
}

message EstablishSessionRequest{
  string name = 1;
}

message EstablishSessionReply{
  bool success = 1;
  string message = 2;
}

message QuestionRequest{
  string name = 1;
  string question = 2;
}

message QuestionReply{
  bool answer = 1;
  string history = 2;
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
